### PR TITLE
bugfix resample only if type is not contour or contourf

### LINKED ISSUE #2537: Rendering problems with add_contours when binary images have rotations set in the affine matrix
**The problem**
When I plot a binary image that has some rotation to it in the affine header using Nilearn's `add_contours` function, the resulting image includes many additional lines that are not part of the surface of the ROI in the binary image. I don't encounter these issues when I use the same code with the same image that has not rotation set in the affine matrix. 

**Reproduction of the issue**
I have attached example data and code to reproduce this problem:
[contour_issue.zip](https://github.com/nilearn/nilearn/files/4871945/contour_issue.zip)

**My environment**
Python 3.8.3 (default, May 27 2020, 20:53:40)
matplotlib==3.2.2
nilearn==0.6.2

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Fixes #2537 

See explanations in #2537. 
Not sure this is the best fix although it's pretty straightforward.

### CODE REVIEW

File: nilearn/plotting/displays.py [hunk 1]
```diff
@@ -750,7 +750,10 @@ def add_contours(self, img, threshold=1e-6, filled=False, **kwargs):
     def _map_show(self, img, type='imshow',
                   resampling_interpolation='continuous',
                   threshold=None, **kwargs):
-        img = reorder_img(img, resample=resampling_interpolation)
+        if _utils.niimg._is_binary_niimg(img):
```
Comment (bthirion) [inline nilearn/plotting/displays.py:753]: Maybe you should explain the logic implemented here.

**Review by bthirion — COMMENTED** [review]
Thx ! You probably want to add a test.

**Comment by NicolasGensollen** [comment]
@bthirion do you think there is still something to improve here ?

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by NicolasGensollen** [comment]
Merging, thanks !

**Pull request merged**